stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -551,7 +551,9 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
             }
             
             if replacementCount != 0 {
-                newElements._copyContents(initializing: bytes + start)
+                let buf = UnsafeMutableBufferPointer(start: bytes + start, count: numericCast(newElements.count))
+                // FIXME: is this guaranteed to return no residual elements?
```

Comment:
You're right, it needs to trap, especially for underflow in this case since it if it doesn't use the entire expected range it will leave corrupt data behind (specific to this being a replaceSubrange use case).